[taniatsq] iP

taniatsq opened the pull request:
## EUEU
> EUEU is your personal ~~chat~~ ***lovebot***! 😍 😍 

He can carry out the following tasks:
- list
- adding todos, deadlines, and events
- mark and unmarking tasks
- deleting tasks

Areas of improvement:
- [ ] GUI

Remember to download Java 11 from [here](https://www.oracle.com/java/technologies/javase/jdk11-archive-downloads.html).

Useful Git commands
``` ruby
git status
git branch
git tag
git commit
```

Top 3 things I learnt:
1. git
2. javafx
3. intellij

The program is run in Duke.class like so: 
```
public class Duke {
    public static void main(String[] args) throws IOException {

          Ui ui = new Ui(user, tasklist);
          System.out.println(ui.showWelcome());
          System.out.println(ui.readCommand());
    }

}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

taniatsq committed:
Add level 0

taniatsq committed:
level 1

taniatsq committed:
level 2

taniatsq committed:
level 3

taniatsq committed:
level 4

taniatsq committed:
level 6

taniatsq committed:
level 6

taniatsq committed:
TextUITesting

taniatsq committed:
level 6

taniatsq committed:
Level 7

taniatsq committed:
Level 8

taniatsq committed:
MoreOOP

taniatsq committed:
MoreOOP

taniatsq committed:
MoreOOP

taniatsq committed:
A-Packages

taniatsq committed:
Merge branch 'add-gradle-support' of https://github.com/taniatsq/ip

taniatsq committed:
Add Gradle

taniatsq committed:
Add JUnit

taniatsq committed:
A-Jar

taniatsq committed:
Level-9

taniatsq committed:
Add level-9

taniatsq committed:
JavaDocs

taniatsq committed:
Merge branch 'Level-9'

Merged with Level 9 and JavaDoc

taniatsq committed:
Add Coding Standards

taniatsq committed:
Add GUI

taniatsq committed:
Add GUI

sun-ruiheng commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,167 @@
+import java.util.Scanner;
+
+import duke.Parser;
+import duke.TaskList;
```
Could be more consistent with the spacing between import lines!

sun-ruiheng commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,167 @@
+    private Image eueu = new Image(this.getClass().getResourceAsStream("/images/DaEUEU.jpg"));
+    public Duke() {
+
+    }
```
If no need for one, I think it's better to leave out the empty constructor

sun-ruiheng replied:
Oops, I just got to the JavaFX. Turns out there's a good reason for this, my mistake!

sun-ruiheng commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,167 @@
+        Ui ui = new Ui(user, tasklist);
+        System.out.println(ui.showWelcome());
+        System.out.println(ui.readCommand());
+
```
Spacing can be more consistent 🤔

sun-ruiheng commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,47 @@
+        this.f = f;
+    }
+
+    /**
```
Nice JavaDoc and Exception handling!

sun-ruiheng commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,200 @@
+    }
+
+    /**
+     * Deletes task in tasklist.
```
Good and consistent JavaDoc for this entire class, and it's nicely organised! 👍

sun-ruiheng commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,77 @@
+     * Returns the String that specifies the Deadline task
+     * @return Deadline category.
+     */
+    public String isDeadline() {
```
"isDeadline" signals that it might be returning a Boolean rather than a String. Perhaps a better name can be given...

sun-ruiheng commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,78 @@
+     * Returns the String that specifies the Event task
+     * @return Event category.
+     */
+    public String isEvent() {
```
Similar to "isDeadline"

sun-ruiheng commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,51 @@
+package duke.task;
```
Very impressive use of inheritance to create each Task subclass! It's neatly organised.

sun-ruiheng reviewed (commented):
Great work overall, Tania! A few nits on code style here and there, like naming conventions, but overall a strong piece of code! 👍

taniatsq committed:
Add GUI

taniatsq committed:
Add D-Contacts

Add new classes Contacts, ContactsList, and ContactsParser.
Contacts: Key fields and features of a contact information to note
ContactsList: Organizes list of contacts
ContactsParser: Parses contact commands to contactlist

taniatsq committed:
Add CodeQuality

Fixed and resolved comments left by peer PR review..
Deleted dead code (unused classes such as UI, MainWindow, Main, etc.).
Replaced magic Strings with named Final constants.

taniatsq committed:
Add A-Assertions

taniatsq committed:
Merge pull request #2 from taniatsq/branch-A-CodeQuality

A-CodeQuality

taniatsq committed:
Merge branch 'master' into branch-A-Assertions

taniatsq committed:
Merge pull request #1 from taniatsq/branch-A-Assertions

Branch-A-Assertions

taniatsq committed:
Add D-Contacts

taniatsq committed:
Add D-Contacts

taniatsq committed:
Merge pull request #4 from taniatsq/branch-D-Contacts

D-Contacts

taniatsq committed:
Edit list method

taniatsq committed:
Edit auto-save
Did not implement save properly before.

I had 2 different lists (one for the current iteration of tasks, and one that was saved in my file), but I could not edit (mark, unmark, delete) the ones that were in the file. Only my current task list was editable.

I made the two list one and the same using one ArrayList tasklist for all tasks.

taniatsq committed:
Add finalized version

taniatsq committed:
Update README.md

taniatsq committed:
Add Ui.png

taniatsq committed:
Add files via upload

taniatsq committed:
Update ContactsParser.java

taniatsq committed:
Merge branch 'master' of https://github.com/taniatsq/ip
Update README.md

taniatsq committed:
Add images

taniatsq committed:
Fix hard coded files

taniatsq committed:
Update bye command

taniatsq committed:
Update CodeQuality

taniatsq committed:
Add blankspace bug fix

taniatsq committed:
Update README.md
